### PR TITLE
Finished Search With Reservation Link

### PR DESCRIPTION
**Pull request by lazpz61BU**
Worked on adding Selenium Testing and added a testing folder with the path of code/Backend/flask-backend/testing. I also left my TDD Testing file in there to test for the HW assignment, as well as my refactored code for the API Endpoints. Was about

### CODE REVIEW

**Review by smathis122 — APPROVED** [review]

File: Pipfile
[the remark this answers is not in the document]
Comment (casejade37) [reply Pipfile:1]: You should probably delete this one and add the other one in .gitignore

File: code/Backend/flask-backend/Pipfile
Comment (casejade37) [inline code/Backend/flask-backend/Pipfile:1]: Should be added into .gitignore

File: code/Backend/flask-backend/app.py [hunk 1]
```diff
@@ -380,71 +386,58 @@ def get_reservation():
...
-        cursor.execute("SELECT * FROM Equipment WHERE available = true")
-        columns = [desc[0] for desc in cursor.description]
-        equipment_data = [dict(zip(columns, row)) for row in cursor.fetchall()]
-
-        cursor.close()
-        conn.close()
 
+        search_query = request.args.get('q')
+        print(f"Received search query: {search_query}")
+        query = "SELECT * FROM Equipment WHERE name ILIKE %s OR description ILIKE %s"
```
Comment (casejade37) [inline code/Backend/flask-backend/app.py:416]: Not sure why you kept using literal string for the query here when you have created query constant on line 34?

File: code/Backend/flask-backend/app.py [hunk 2]
```diff
@@ -380,71 +386,58 @@ def get_reservation():
...
-        columns = [desc[0] for desc in cursor.description]
-        equipment_data = [dict(zip(columns, row)) for row in cursor.fetchall()]
-
-        cursor.close()
-        conn.close()
-
+        availability = request.args.get('availability')
+        query = "SELECT * FROM Equipment"
+        if availability == "available":
+            query += " WHERE available = true"
```
Comment (casejade37) [inline code/Backend/flask-backend/app.py:433]: Same question here: why create another query variable when you have created constants on line 35-36?

File: code/Backend/flask-backend/app.py (hunk 1 above)
Comment (lazpz61BU) [reply code/Backend/flask-backend/app.py:481]: I forgot to update my backend code for that, I had been juggling trying to get Selenium to work, and finish up researching how to pass the item to the Reservation Page I Missed to make the change

File: code/Backend/flask-backend/app.py (hunk 2 above)
Comment (lazpz61BU) [reply code/Backend/flask-backend/app.py:498]: Same as my last comment.

**Review by casejade37 — APPROVED** [review]
Good job on the updates!

**Pull request merged**